src/app/sidenav/sidenav.component.ts

```diff
@@ -9,40 +13,60 @@ import {Utils} from '../utils';
 })
 export class SidenavComponent implements OnInit {
   @ViewChild('sidenav') sidenav: MatSidenav;
-  communities: Array<any> = [];
-  communitiesFavourited: Array<any> = [];
+  communities: Array<Community> = [];
+  communitiesFavourited: Array<Community> = [];
 
   get sidebarWidth(): number {
     return 300;
   }
 
   utils = Utils;
 
-  constructor() { }
+  constructor(private _UserService: UserService, private _CommunityService: CommunityService) { }
 
   ngOnInit() {
-    // populate 20 random communities
-    for (let i = 0; i < 20; i++) {
-      this.communities.push({ communityname: 'Community ' + i,
-      isFavourited: false,
-      favouriteCount: Math.floor(Math.random() * (101))});
-    }
-    for (const community of this.communities) {
-      if (community.isFavourited === true) { this.communitiesFavourited.push(community); }
-    }
+
+    this._CommunityService.getCommunities('top', 10, 0)
```

Comment:
I'm little bit confused as to what the `resolve` and `reject` functions do here from the `q`. If you want, you can use async/await to make the promise code appear like synchronous stuff. 

```javascript
async ngOnInit() {

    let communityList = await this._CommunityService.getCommunities('top', 10, 0);


    if (this._UserService.isLoggedIn()) {
      // Display user communities
      const user = await this._UserService.getDetails(true);

      // Remove items in the top community list that are in their favourites already
      communityList.communities = communityList.communities.filter((community) => {
        return user.Favourites.findIndex((c) => c.name === community.name) === -1;
      });

      this.communitiesFavourited = user.Favourites as Community[];
    }
    

    this.communities = communityList.communities;
}
```

Reply:
Okay I see. I wasn't sure how to use the API services since they returned promises, which I assumed could only be dealt with using `resolve` and `reject`.